### ISSUE TITLE
fails to compile with gcc 4.8, compiles with 4.9

### ISSUE THREAD

**Issue by zetok** [qTox/qTox#1699]
OS: Hardened Gentoo Linux x86_64
qTox: a841224683116e75852d6d2ebc9a9f1dc70ec957
`gcc-4.8.4` version: gcc (Gentoo Hardened 4.8.4 p1.4, pie-0.6.1) 4.8.4
`gcc-4.9.2` version: gcc (Gentoo Hardened 4.9.2 p1.3, pie-0.6.2) 4.9.2

@Zer0-One ran into that issue, and I was able to confirm that qTox compiles successfully with `gcc 4.9.2`, while it fails with `gcc 4.8.4`.

Relevant output with `gcc 4.8.4`:

```
x86_64-pc-linux-gnu-g++ -c -ggdb -O1 -pipe -Wall -Wextra -pedantic -Wformat -Wformat-security -std=c++0x -Wall -W -D_REENTRANT -fPIE -DGIT_VERSION="\"a841224683116e75852d6d2ebc9a9f1dc70ec957\"" -DTIMESTAMP=1432159588 -DLOG_TO_FILE -DQT_MESSAGELOGCONTEXT -DQTOX_PLATFORM_EXT -DQTOX_FILTER_AUDIO -DQT_OPENGL_LIB -DQT_SVG_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_XML_LIB -DQT_SQL_LIB -DQT_CORE_LIB -I. -Ilibs/include -isystem /usr/include/qt5 -isystem /usr/include/qt5/QtOpenGL -isystem /usr/include/qt5/QtSvg -isystem /usr/include/qt5/QtWidgets -isystem /usr/include/qt5/QtGui -isystem /usr/include/qt5/QtNetwork -isystem /usr/include/qt5/QtXml -isystem /usr/include/qt5/QtSql -isystem /usr/include/qt5/QtCore -I. -I. -I/usr/lib64/qt5/mkspecs/linux-g++ -o ipc.o src/ipc.cpp
x86_64-pc-linux-gnu-g++ -c -ggdb -O1 -pipe -Wall -Wextra -pedantic -Wformat -Wformat-security -std=c++0x -Wall -W -D_REENTRANT -fPIE -DGIT_VERSION="\"a841224683116e75852d6d2ebc9a9f1dc70ec957\"" -DTIMESTAMP=1432159588 -DLOG_TO_FILE -DQT_MESSAGELOGCONTEXT -DQTOX_PLATFORM_EXT -DQTOX_FILTER_AUDIO -DQT_OPENGL_LIB -DQT_SVG_LIB -DQT_WIDGETS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_XML_LIB -DQT_SQL_LIB -DQT_CORE_LIB -I. -Ilibs/include -isystem /usr/include/qt5 -isystem /usr/include/qt5/QtOpenGL -isystem /usr/include/qt5/QtSvg -isystem /usr/include/qt5/QtWidgets -isystem /usr/include/qt5/QtGui -isystem /usr/include/qt5/QtNetwork -isystem /usr/include/qt5/QtXml -isystem /usr/include/qt5/QtSql -isystem /usr/include/qt5/QtCore -I. -I. -I/usr/lib64/qt5/mkspecs/linux-g++ -o autoupdate.o src/autoupdate.cpp
src/ipc.cpp: In constructor ‘IPC::IPC()’:
src/ipc.cpp:37:65: error: ‘type name’ declared as function returning a function
     std::default_random_engine randEngine((std::random_device()()));
                                                                 ^
src/ipc.cpp:37:65: error: ‘type name’ declared as function returning a function
Makefile:2481: recipe for target 'ipc.o' failed
make: *** [ipc.o] Error 1
make: *** Waiting for unfinished jobs....
```

**Comment by tux3** [comment]
This is a bug in GCC: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=29234

I'll push a patch to work around it later today. We can just do that in two lines.

**Comment by zetok** [comment]
Fixed by f71602718025779ce81554baa822ccaa551c10e2
